ofctrl/fgraphSwitch.go

```diff
@@ -96,31 +97,39 @@ func (self *OFSwitch) DefaultTable() *Table {
 	return self.tableDb[0]
 }
 
-// Create a new group. return an error if it already exists
-func (self *OFSwitch) NewGroup(groupId uint32, groupType GroupType) (*Group, error) {
-	// check if the group already exists
-	if self.groupDb[groupId] != nil {
-		return nil, errors.New("group already exists")
+// NewGroup creates a new group; when using cache, returns an error if the group ID already exists, otherwise returns the
+// group object; when not using cache, returns the new group object， and the caller should ensure the groupID is not
+// duplicated.
+func (self *OFSwitch) NewGroup(groupID uint32, groupType GroupType, useCache bool) (*Group, error) {
+	// Check if the group already exists.
+	if useCache {
+		if self.groupDb[groupID] != nil {
+			return nil, errors.New("group already exists")
+		}
 	}
 
-	// Create a new group
-	group := newGroup(groupId, groupType, self)
-	// Save it in the DB
-	self.groupDb[groupId] = group
+	// Create a new group.
+	group := newGroup(groupID, groupType, self)
+	if useCache {
+		// Save it in cache.
+		self.groupDb[groupID] = group
+	}
 
 	return group, nil
 }
 
-// Delete a group.
-// Return an error if there are flows refer pointing at it
+// DeleteGroup deletes a group in cache.
 func (self *OFSwitch) DeleteGroup(groupId uint32) error {
 	delete(self.groupDb, groupId)
 	return nil
 }
 
-// GetGroup Returns a group
-func (self *OFSwitch) GetGroup(groupId uint32) *Group {
-	return self.groupDb[groupId]
+// GetGroup returns a group if it is cached.
```

Comment:
Add comment to explain that this function is used only when the queried group is created using the cache?

Reply:
Done.